libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -1013,7 +1013,9 @@ void AP_Baro::update_field_elevation(void)
             is_zero(_field_elevation)) {
             // auto-set based on origin
             Location origin;
-            if (!armed && AP::ahrs().get_origin(origin)) {
+            if (!armed &&
+                AP::ahrs().get_origin(origin) &&
+                origin.alt) {
```

Comment:
Maybe it's a personal style thing but I prefer not to treats integers as booleans.  so I'd prefer if the additional check was, (origin.alt != 0);